drivers: stepper: add invert direction to gpio stepper

jilaypandya opened the pull request:
This commit implements invert direction feature in gpio stepper

bjarki-andreasen commented:
looks good, but could you split the renaming of the compatible, and the inclusion of the new property into separate commits?

decsny commented:
you don't need to fight the bot, it puts DT label for intended reason

jilaypandya committed:
drivers: stepper: rename compatible of gpio-stepper

This commit fixes minor copyright issues and corrects the compatible of
gpio-stepper with the vendor name as zephyr

Signed-off-by: Jilay Pandya <jilay.pandya@outlook.com>

jilaypandya committed:
drivers: stepper: introduce invert-direction property to gpio-stepper

This commit introduces invert-direction property to gpio-stepper

Signed-off-by: Jilay Pandya <jilay.pandya@outlook.com>

dipakgmx commented on drivers/stepper/gpio_stepper_controller.c:
```diff
@@ -326,6 +345,7 @@ static int gpio_stepper_motor_controller_init(const struct device *dev)
 		ARRAY_SIZE(gpio_stepper_motor_control_pins_##child) == 4,                          \
 		"gpio_stepper_controller driver currently supports only 4 wire configuration");    \
 	static const struct gpio_stepper_config gpio_stepper_config_##child = {                    \
+		.invert_direction = DT_PROP(child, invert_direction),                              \
```
```suggestion
		.invert_direction = DT_PROP_OR(child, invert_direction, 0),                              \
```

jilaypandya replied:
Boolean properties are true if present. They should not have a value. A boolean property is only false if it is completely missing in the DTS.

bjarki-andreasen reviewed (dismissed):
nice :)

fabiobaltieri commented on drivers/stepper/gpio_stepper_controller.c:
```diff
@@ -53,20 +55,35 @@ static int stepper_motor_set_coil_charge(const struct device *dev)
+{
+	struct gpio_stepper_data *data = dev->data;
+
+	data->coil_charge =
+		data->coil_charge == NUM_CONTROL_PINS * MAX_MICRO_STEP_RES - data->step_gap
+			? 0
+			: data->coil_charge + data->step_gap;
```
yikes, can you expand it to a normal `if`? the indentation is something else here

fabiobaltieri commented on drivers/stepper/gpio_stepper_controller.c:
```diff
@@ -53,20 +55,35 @@ static int stepper_motor_set_coil_charge(const struct device *dev)
+{
+	struct gpio_stepper_data *data = dev->data;
+
+	data->coil_charge = data->coil_charge == 0
+				    ? NUM_CONTROL_PINS * MAX_MICRO_STEP_RES - data->step_gap
+				    : data->coil_charge - data->step_gap;
```
same
